nixos/modules/profiles/hardened.nix

```diff
@@ -7,7 +7,7 @@ with lib;
 
 {
   meta = {
-    maintainers = [ maintainers.joachifm ];
+    maintainers = [ maintainers.joachifm maintainers.emily ];
```

Comment:
Maybe one should mention in `meta.doc = ./hardened.xml;` that in order to use Firefox with this profile, one has to reset the memory allocator:

https://github.com/NixOS/nixpkgs/pull/73763#issuecomment-614903630

cc @Flakebi

Reply:
Agreed; not sure it belongs in this PR which is entirely kernel changes and already huge, though, as Firefox is broken with a custom allocator set regardless of what kernel you use. I think the correct solution to make Firefox work is to implement a library with [replace-malloc hooks](https://searchfox.org/mozilla-central/rev/567b68b8ff4b6d607ba34a6f1926873d21a7b4d7/memory/build/replace_malloc.h); I've been idly planning to do that since the scudo change got merged but I'd be very happy for someone to preempt me! It'd also be nice to have support for setting up Firefox's setuid sandbox like Chromium's if we're disabling unprivileged userns.

FWIW, from playing around with things I think the `"graphene-hardened"` allocator might have better compatibility for some programs than scudo (and it's what I was planning on targetting with the replace-malloc library), but ultimately I suspect that we're going to want a way to specifically exempt packages from the hardened allocator. With the current implementation that involves mount namespaces so you can hide `/etc/ld-nix.so.preload`, which combined with the userns defaults means that the wrappers to do it will need to be setuid root :(